hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/FileIndex.java

```diff
@@ -83,8 +77,8 @@ private FileIndex(StoragePath path, Configuration conf, RowType rowType, DataPru
     this.rowType = rowType;
     this.hadoopConf = HadoopConfigurations.getHadoopConf(conf);
     this.tableExists = StreamerUtil.tableExists(path.toString(), hadoopConf);
-    this.metadataConfig = metadataConfig(conf);
-    this.dataPruner = isDataSkippingFeasible(conf.getBoolean(FlinkOptions.READ_DATA_SKIPPING_ENABLED)) ? dataPruner : null;
+    this.metadataConfig = StreamerUtil.metadataConfig(conf);
+    this.dataPruner = isDataSkippingFeasible(conf.get(FlinkOptions.READ_DATA_SKIPPING_ENABLED)) ? dataPruner : null;
```

Comment:
Maybe we should rename the pruner to `colStatsPruner` because it is used for partition pruning now.

Reply:
Let's rename the `DatatPruner` to `ColStatsProbe`.